### ISSUE TITLE
FR: EPUB Navigation Landmarks/Guide support

### ISSUE THREAD

**Issue by Kayzels** [koreader/koreader#8509]
This is just an idea, not sure whether it would be tricky to implement. EPUB2 books have a `guide` section in the .opf file that describes different semantic parts of a book, such as the cover, copyright-page and so on. EPUB3 has a similar section in the nav file, called `landmarks`.

I was wondering whether this could be exposed as another option in the Table of Contents menu.
This would be useful when there is a lot of frontmatter, and you just want to start reading the book, because there is the item `text` in the opf file, which was renamed to `bodymatter` in the nav file, which refers to the place where the actual content starts.
As there is already support for Reference Pages, which uses the `page-list` nav item (see below), I was wondering whether exposing the `landmarks` nav item would be possible.
![Reader_2021-12-04_095751](https://user-images.githubusercontent.com/72696811/144702444-39559f6a-5447-4fa3-af90-2c3317c496b9.png)

I have attached the nav file (as a zip file, because github doesn't support xhtml files), for the same file in the screenshot, which is a copyrighted work. If that is not allowed, please let me know.
[nav.zip](https://github.com/koreader/koreader/files/7653580/nav.zip)

**Comment by Frenzie** [comment]
Please use [ScrambleEbook](https://www.mobileread.com/forums/showthread.php?t=267998&highlight=epub+scrambler).

**Comment by poire-z** [comment]
It might not be super tricky nor impossible, but it's a bit heavy on the crengine side (needs caching these in the crengine book cache) - would be similar to what we added for reference pages: https://github.com/koreader/crengine/pull/336/commits/e5d89aa9b5222b70ede51c0c257e563dc75ad40b

But I personally think this "landmarks" feature is a bit useless and redundant, and I would mind another useless menu item in the first top level menu.
In your sample (and in ALL the books I looked at when looking at nav.xml files when adding reference pages), we have very little interesting things:
```xml
    <nav epub:type="landmarks">
      <h2 class="contentsHead">Guide</h2>
      <ol>
        <li><a href="cover.xhtml" epub:type="cover">Cover</a></li>
        <li><a href="titlepage.xhtml" epub:type="titlepage">Title Page</a></li>
        <li><a href="dedication.xhtml" epub:type="dedication">Dedication</a></li>
        <li><a href="nav.xhtml#toc" epub:type="toc">Contents</a></li>
        <li><a href="copyright.xhtml" epub:type="copyright-page">Copyright</a></li>
        <li><a href="part01.xhtml" epub:type="bodymatter">Begin Reading</a></li>
      </ol>
    </nav>
```
which just flags a few of the early pages, which are also all mentionned in the TOC:
```xml
    <nav epub:type="toc" id="toc">
      <h2 class="contentsHead">Contents</h2>
      <ol>
        <li><a href="cover.xhtml">Cover</a></li>
        <li><a href="praise.xhtml">Praise</a></li>
        <li><a href="also_by.xhtml">Also by Jeffrey Archer</a></li>
        <li><a href="titlepage.xhtml">Title Page</a></li>
        <li><a href="dedication.xhtml">Dedication</a></li>
        <li><a href="nav.xhtml#toc">Contents</a></li>
[...]
        <li class="space"><a href="copyright.xhtml">Copyright</a></li>
      </ol>
    </nav>
```
Moreover, very often, this nav OL/LI (or some other copy of that) is included in the book content (and so is the OL/LI from the TOC), which is just a lot of crap I skip manually.

> This would be useful when there is a lot of frontmatter, and you just want to start reading the book, because there is the item text in the opf file, which was renamed to bodymatter in the nav file, which refers to the place where the actual content starts.

Yes, we have to skip the crap. But I'm not in favor of skipping it automatically: It's mostly always at the start or at the end, so we usually quickly skip them when opening a new book - but I think that is just normal (like in real paper books): I'd rather not miss "praise" and "dedication" because I'm jumped to chapter1 - it's just 20 second of my life skipping them manually :) but being sure I didn't miss valuable book content.

**Comment by Kayzels** [comment]
I agree that actually having the whole landmarks area is extra information, so adding the entire list is not a great idea. My main reasoning for this feature request is that when you read a book on the Kindle, there is an entry in the "Go To" page before the Table of Contents, that says "Beginning", which goes to the marked start. But I don't like that the Kindle automatically goes there.

So, a more reasonable proposal would be to not have the entire landmarks list, but have an entry marked like "Go to Start" which would go to the marked text start. Opening the book would not automatically go there, but just expose that as an option on the Navigation Menu (maybe even in the same section as Reference Pages).
